### DISCUSSION TITLE
Creating a superuser with one command

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2617] [Community Contribs & Snippets]
As you know, setting up Evennia is done with 

    evennia migrate 
    evennia start 

When you run `evennia start` for the first time you will be asked to create a superuser in a few interactive steps. If you want to automate this process (such as for automatic deployment) this won't work. You can do this by adding the following between the `migrate` and `start` steps (*nix command line): 

    echo "from evennia import AccountDB; AccountDB.objects.create_superuser('admin', 'admin@myproject.com', 'password')" | evennia shell

or with envvars: 

    echo "from evennia import AccountDB; AccountDB.objects.create_superuser('$USERNAME', '$EMAIL', '$PASSWORD')" | evennia shell

Note that you will not have a Character when logging in, you'll have to create that manually.

**Comment by Russell-Jones-OxPhys** [comment]
Hi Griatch, I tried this in an OpenShift container (I'm using a rather sketchy Dockerfile built on the one supplied with Evennia and putting the game folder in /tmp as proof of concept) and I get an infinite loop of
```
Create a superuser below. The superuser is Account #1, the 'owner' account of the server. Email is optional and can be empty.
Superuser creation skipped due to not running in a TTY. You can run `manage.py createsuperuser` in your project to create one manually.
```
My repo has a default gamedir as described in the Evennia dockerfile
I run evennia --initmissing (for a fresh settings_secret.py) and evennia migrate

I see this if I run evennia start -l as well.

**Comment by atebyagrue** [comment]
I automate Django admin permission creation in Docker, but I am unfamiliar with OpenShift. It took me a while to get it sorted, but to do it in Docker, you need to set your environment variables (DJANGO_SUPERUSER_USERNAME, DJANGO_SUPERUSER_PASSWORD, and DJANGO_SUPERUSER_EMAIL) and then execute "python manage.py createsuperuser --noinput" from within the container. I spent a lot of cycles getting this to work & hope this saves you some time getting an answer.
